lib/puppet_blacksmith/forge.rb

```diff
@@ -51,15 +51,35 @@ def push!(name, package = nil, author = nil, version = nil)
     private
 
     def upload(author, name, file)
-      url = http_url(author, name, file)
-      case forge_type
-      when FORGE_TYPE_ARTIFACTORY
-        RestClient::Request.execute(method: :put, url: url, payload: File.new(file, 'rb'), headers: http_headers)
+      target_url = http_url(author, name, file)
+      uri = URI(target_url)
+
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = (uri.scheme == 'https')
+
+      if forge_type == FORGE_TYPE_ARTIFACTORY
+        request = Net::HTTP::Put.new(uri.request_uri, http_headers)
+        request.body_stream = File.open(file, 'rb')
       else
-        RestClient::Request.execute(method: :post, url: url, payload: { file: File.new(file, 'rb') }, headers: http_headers)
+        request = Net::HTTP::Post.new(uri.request_uri, http_headers)
+        request.set_form([
+                           ['file', File.open(file, 'rb'), { filename: File.basename(file) }],
+                         ], 'multipart/form-data')
       end
-    rescue RestClient::Exception => e
-      raise Blacksmith::Error, "Error uploading #{name} to the forge #{url} [#{e.message}]: #{e.response}"
+
+      begin
+        response = http.request(request)
+      ensure
+        request.body_stream&.close if request.body_stream.is_a?(File)
+      end
+
+      raise Blacksmith::Error, "Error uploading #{name} to the forge #{target_url} [#{response.code} #{response.message}]: #{response.body}" unless response.code.to_i >= 200 && response.code.to_i < 300
```

Comment:
You can also check the instance which is a bit easier to read:
```suggestion
      raise Blacksmith::Error, "Error uploading #{name} to the forge #{target_url} [#{response.code} #{response.message}]: #{response.body}" unless response.is_a?(Net::HTTPSuccess)
```